src/tracer.js

```diff
@@ -202,29 +202,40 @@ function getTrimmedMetadata(eventMetadata, isRunner) {
 function getTrimmedTrace(traceSize, jsTrace) {
```

Comment:
this function is becoming too long (and too nested). Consider moving every "if" block content to its own dedicated function (`trimTraceExceptions`, `trimTraceEventsMetadata`, `trimTraceEvents`, etc)